src/condor/htcondor_helper.py

```diff
@@ -2,174 +2,332 @@
 import time
 import os
 from typing import Any
-from condor.utils import setup_logger
-
-# Get the centralized logger instance
-logger = setup_logger()
 
 class HTCondorHelper:
-    def __init__(self, config: 'dict[str, Any]'):
-        """Initialize HTCondor helper with configuration"""
+    def __init__(self, config, token_dir: str):
+        """Initialize HTCondor helper with Config object and token directory"""
         self.config = config
-        self.token = config["token"]
-        self.token_file = config["token_file"]
-        # Remove debug logging
-        # htcondor.enable_debug()
+        self.logger = config.logger
+        self.token_dir = token_dir
+        self._validate_config()
         self._setup_connection()
-        logger.info("HTCondor helper initialized")
+
+    def _validate_config(self):
+        """Validate required HTCondor configuration"""
+        required_attrs = [
+            'collector_host',
+            'submit_host',
+            'docker_image',
+            'request_gpus',
+            'request_cpus',
+            'request_memory',
+            'request_disk'
+        ]
+        
+        missing = [attr for attr in required_attrs if not hasattr(self.config, attr)]
+        if missing:
+            raise ValueError(f"Missing HTCondor config attributes: {', '.join(missing)}")
 
     def _setup_connection(self):
         """Setup connection to HTCondor submit node"""
         try:
-            if not os.path.exists(self.token_file):
-                raise ValueError(f"Token file not found at {self.token_file}")
+            # Print info about the token directory
+            self.logger.info(f"Token directory: {self.token_dir}")
+            if os.path.exists(self.token_dir):
```

Comment:
is there a reason to use both token_dir and token_file? can we just use token_file? (please review the code that this PR removes)